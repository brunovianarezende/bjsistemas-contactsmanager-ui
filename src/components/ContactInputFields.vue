<template>
  <el-form ref="form" :model="contactObj" :rules="rules">
    <el-form-item label="First name" prop="firstname">
      <el-input v-model="contactObj.firstname"></el-input>
    </el-form-item>
    <el-form-item label="Last name" prop="lastname">
      <el-input v-model="contactObj.lastname"></el-input>
    </el-form-item>
    <el-form-item label="Date of birth" prop="birthDate">
      <el-date-picker
        v-model="contactObj.birthDate"
        type="date"
        format="dd/MM/yyyy"
        formatValue="yyyy-MM-dd"
        placeholder="Pick a day">
      </el-date-picker>
    </el-form-item>
    <el-form-item prop="emails">
      <label class="el-form-item__label no-float">E-mails</label>
      <el-form-item
        v-for="(email, index) in contactObj.emails"
        :key="index"
        :prop="`emails.${index}`"
        :rules="emailRules"
      >
        <el-input class="email-input" v-model="contactObj.emails[index]"></el-input>
        <span class="delete-item" v-on:click="onRemoveEmail(index)"><icon scale="2" name="minus-circle" color="red" /></span>
      </el-form-item>
      <div class="add-item" v-on:click="onAddEmail">
        <icon name="plus-circle" scale="2" color="green" />
      </div>
    </el-form-item>
    <fieldset>
      <legend>Addresses</legend>
      <template
        v-for="(address, index) in contactObj.addresses"
      >
        <div :key="index">
          <span class="delete-address" v-on:click="onRemoveAddress(index)"><icon name="minus-circle" scale="2" color="red" /></span>
          <edit-address :address="address" :base-prop-id="`addresses.${index}`" />
        </div>
      </template>
      <div class="add-item" v-on:click="onAddAddress">
        <icon name="plus-circle" scale="2" color="green" />
      </div>
    </fieldset>
    <el-form-item prop="phoneNumbers">
      <label class="el-form-item__label no-float">Phone numbers</label>
      <el-form-item
        v-for="(phoneNumber, index) in contactObj.phoneNumbers"
        :key="index"
        :prop="`phoneNumbers.${index}`"
        :rules="phoneNumberRules"
      >
        <el-input class="email-input" v-model="contactObj.phoneNumbers[index]"></el-input>
        <span class="delete-item" v-on:click="onRemovePhoneNumber(index)"><icon scale="2" name="minus-circle" color="red" /></span>
      </el-form-item>
    <div class="add-item" v-on:click="onAddPhoneNumber">
      <icon name="plus-circle" scale="2" color="green" />
    </div>
    </el-form-item>
  </el-form>
</template>

<script>
import EditAddress from './EditAddress'
import 'vue-awesome/icons/plus-circle'
import 'vue-awesome/icons/minus-circle'
import Icon from 'vue-awesome/components/Icon'

export default {
  name: 'ContactInputFields',
  props: ['contact'],
  data () {
    const contactObj = this.contact

    return {
      contactObj,
      rules: {
        firstname: { required: true, whitespace: true, message: 'First name is required', trigger: 'blur' },
        lastname: { required: true, whitespace: true, message: 'Last name is required', trigger: 'blur' },
        birthDate: { required: true, type: 'object', message: 'Date of birth is required', trigger: 'blur' },
        emails: {
          required: true,
          type: 'array',
          min: 1,
          message: 'Add at least one email',
          trigger: 'blur'
        },
        phoneNumbers: { required: true, type: 'array', min: 1, message: 'Add at least one phone number', trigger: 'blur' }
      },
      emailRules: [
        { whitespace: true, required: true, message: 'Email address is required', trigger: 'blur' },
        { type: 'email', message: 'Email address must be valid', trigger: 'blur' }
      ],
      phoneNumberRules: [
        { whitespace: true, required: true, message: 'Phone number is required', trigger: 'blur' },
      ]
    }
  },
  methods: {
    getForm () {
      return this.$refs.form
    },
    onAddAddress () {
      this.contactObj.addresses = [
        ...this.contactObj.addresses,
        {street: '', city: '', state: '', zipCode: ''}
      ]
    },
    onRemoveAddress (indexToRemove) {
      this.contactObj.addresses = this.contactObj.addresses
        .filter((i, index) => index !== indexToRemove)
    },
    onAddEmail () {
      this.contactObj.emails = [
        ...this.contactObj.emails,
        ''
      ]
    },
    onRemoveEmail (indexToRemove) {
      this.contactObj.emails = this.contactObj.emails
        .filter((i, index) => index !== indexToRemove)
    },
    onAddPhoneNumber () {
      this.contactObj.phoneNumbers = [
        ...this.contactObj.phoneNumbers,
        ''
      ]
    },
    onRemovePhoneNumber (indexToRemove) {
      this.contactObj.phoneNumbers = this.contactObj.phoneNumbers
        .filter((i, index) => index !== indexToRemove)
    }
  },
  components: {
    EditAddress, Icon
  }
}
</script>

<style lang="scss" scoped>
.add-item {
  margin: 10px;
}
.delete-item {
  margin: 10px;
}
.delete-address {
  position: relative;
  left: 0px;
  top: 15px;
}
.email-input {
  width: 40%;
}
.no-float {
  float: none;
}
.el-form-item .el-form-item {
  margin-bottom: 22px;
}
</style>
