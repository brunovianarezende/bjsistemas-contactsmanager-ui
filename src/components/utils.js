export const notEmpty = (l) => l.filter((i) => !!i)

export const formatAddress = (address, separator) => {
  const lines = [
    formatFirstLine(address),
    formatSecondLine(address)
  ]
  return notEmpty(lines).join(separator)
}

export const formatFirstLine = (address) => {
  return address.street || ''
}

export const formatSecondLine = (address) => {
  return notEmpty([address.city, address.state, address.zipCode]).join(', ')
}

const padStart = (n) => {
  return ('0' + n).slice(-2)
}

export const formatDate = (date) => {
  const dateObj = new Date(date)
  if (!isNaN(dateObj.getDate())) {
    dateObj.setMinutes(dateObj.getTimezoneOffset())
    return `${padStart(dateObj.getDate())}/${padStart(dateObj.getMonth() + 1)}/${dateObj.getFullYear()}`
  } else {
    return ''
  }
}
