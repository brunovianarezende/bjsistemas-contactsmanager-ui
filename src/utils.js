import Vue from 'vue'
import ElementUI from 'element-ui'
import VModal from 'vue-js-modal'

export const registerGlobalComponents = () => {
  Vue.use(ElementUI)
  Vue.use(VModal)
}
