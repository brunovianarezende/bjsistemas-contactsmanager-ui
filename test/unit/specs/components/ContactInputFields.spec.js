import Vue from 'vue'
import ContactInputFields from '@/components/ContactInputFields'
import {registerGlobalComponents} from '@/utils'

registerGlobalComponents()

const deepClone = (o) => {
  return JSON.parse(JSON.stringify(o))
}

describe('ContactInputFields.vue', () => {
  it('should render the content correctly when no contact is given', () => {
    const Constructor = Vue.extend(ContactInputFields)
    const emptyContact = {
      firstname: '',
      lastname: '',
      addresses: [],
      emails: [],
      phoneNumbers: []
    }
    const vm = new Constructor({propsData: {
      contact: emptyContact
    }}).$mount()
    return Vue.nextTick().then(() =>
      expect(vm.$el).toMatchSnapshot()
    )
  })

  it('should render the content correctly when a contact is given', () => {
    const contact = {
      firstname: 'First',
      lastname: 'Last',
      addresses: [{street: '1373 George Avenue', city: 'Montgomery', state: 'AL', zipCode: '36693'}],
      emails: ['email1@email.com', 'email2@email.com'],
      phoneNumbers: ['55-31-99967-0000', '55-31-2515-0000']
    }
    const Constructor = Vue.extend(ContactInputFields)
    const vm = new Constructor({ propsData: {contact} }).$mount()
    return Vue.nextTick().then(() =>
      expect(vm.$el).toMatchSnapshot()
    )
  })

  it('should add an address if \'add address\' is clicked', () => {
    const contact = {id: 1, firstname: 'Bruno', lastname: 'Rezende', birthDate: '1980-03-12', addresses: [{street: '1373 George Avenue', city: 'Montgomery', state: 'AL', zipCode: '36693'}], emails: ['brunovianarezende@gmail.com'], 'phoneNumbers': ['55-31-2515-5924', '55-31-99967-7424']}

    const Constructor = Vue.extend(ContactInputFields)
    const vm = new Constructor({propsData: {contact: deepClone(contact)}}).$mount()

    return Vue.nextTick()
      .then(() => {
        vm.onAddAddress()
        expect(vm.contactObj.addresses).toEqual([
          ...contact.addresses,
          {street: '', city: '', state: '', zipCode: ''}
        ])
      })
  })

  it('should remove an address if \'remove address\' is clicked', () => {
    const a = (prefix) => {
      return {street: `${prefix} - 1373 George Avenue`, city: 'Montgomery', state: 'AL', zipCode: '36693'}
    }

    const contact = {
      id: 1,
      firstname: 'Bruno',
      lastname: 'Rezende',
      birthDate: '1980-03-12',
      addresses: [a(1), a(2), a(3)],
      emails: ['brunovianarezende@gmail.com'],
      phoneNumbers: ['55-31-2515-5924', '55-31-99967-7424']
    }

    const Constructor = Vue.extend(ContactInputFields)
    const vm = new Constructor({propsData: {contact: deepClone(contact)}}).$mount()

    return Vue.nextTick()
      .then(() => {
        vm.onRemoveAddress(1)
        expect(vm.contactObj.addresses).toEqual([
          a(1), a(3)
        ])
      })
  })

  it('should add an email if \'add email\' is clicked', () => {
    const contact = {id: 1, firstname: 'Bruno', lastname: 'Rezende', birthDate: '1980-03-12', addresses: [{street: '1373 George Avenue', city: 'Montgomery', state: 'AL', zipCode: '36693'}], emails: ['brunovianarezende@gmail.com'], 'phoneNumbers': ['55-31-2515-5924', '55-31-99967-7424']}

    const Constructor = Vue.extend(ContactInputFields)
    const vm = new Constructor({propsData: {contact: deepClone(contact)}}).$mount()

    return Vue.nextTick()
      .then(() => {
        vm.onAddEmail()
        expect(vm.contactObj.emails).toEqual([
          ...contact.emails,
          ''
        ])
      })
  })

  it('should remove an email if \'remove email\' is clicked', () => {
    const e = (prefix) => `${prefix}email@email.com`

    const contact = {
      id: 1,
      firstname: 'Bruno',
      lastname: 'Rezende',
      birthDate: '1980-03-12',
      addresses: [{street: '1373 George Avenue', city: 'Montgomery', state: 'AL', zipCode: '36693'}],
      emails: [e(1), e(2), e(3)],
      phoneNumbers: ['55-31-2515-5924', '55-31-99967-7424']
    }

    const Constructor = Vue.extend(ContactInputFields)
    const vm = new Constructor({propsData: {contact: deepClone(contact)}}).$mount()

    return Vue.nextTick()
      .then(() => {
        vm.onRemoveEmail(1)
        expect(vm.contactObj.emails).toEqual([
          e(1), e(3)
        ])
      })
  })

  it('should add a phone number if \'add phone number\' is clicked', () => {
    const contact = {id: 1, firstname: 'Bruno', lastname: 'Rezende', birthDate: '1980-03-12', addresses: [{street: '1373 George Avenue', city: 'Montgomery', state: 'AL', zipCode: '36693'}], emails: ['brunovianarezende@gmail.com'], 'phoneNumbers': ['55-31-2515-5924', '55-31-99967-7424']}

    const Constructor = Vue.extend(ContactInputFields)
    const vm = new Constructor({propsData: {contact: deepClone(contact)}}).$mount()

    return Vue.nextTick()
      .then(() => {
        vm.onAddPhoneNumber()
        expect(vm.contactObj.phoneNumbers).toEqual([
          ...contact.phoneNumbers,
          ''
        ])
      })
  })

  it('should remove a phone number if \'remove phone number\' is clicked', () => {
    const p = (prefix) => `${prefix}5-31-1234-5678`

    const contact = {
      id: 1,
      firstname: 'Bruno',
      lastname: 'Rezende',
      birthDate: '1980-03-12',
      addresses: [{street: '1373 George Avenue', city: 'Montgomery', state: 'AL', zipCode: '36693'}],
      emails: [],
      phoneNumbers: [p(1), p(2), p(3)]
    }

    const Constructor = Vue.extend(ContactInputFields)
    const vm = new Constructor({propsData: {contact: deepClone(contact)}}).$mount()

    return Vue.nextTick()
      .then(() => {
        vm.onRemovePhoneNumber(1)
        expect(vm.contactObj.phoneNumbers).toEqual([
          p(1), p(3)
        ])
      })
  })
})
