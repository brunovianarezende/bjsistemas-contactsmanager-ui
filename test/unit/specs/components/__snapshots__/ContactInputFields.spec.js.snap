// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactInputFields.vue should render the content correctly when a contact is given 1`] = `
<form
  class="el-form"
>
  <div
    class="el-form-item is-required"
  >
    <label
      class="el-form-item__label"
      for="firstname"
    >
      First name
    </label>
    <div
      class="el-form-item__content"
    >
      <div
        class="el-input"
      >
        <!---->
        <input
          autocomplete="off"
          class="el-input__inner"
          rows="2"
          type="text"
          validateevent="true"
        />
        <!---->
        <!---->
        <!---->
      </div>
      <!---->
    </div>
  </div>
   
  <div
    class="el-form-item is-required"
  >
    <label
      class="el-form-item__label"
      for="lastname"
    >
      Last name
    </label>
    <div
      class="el-form-item__content"
    >
      <div
        class="el-input"
      >
        <!---->
        <input
          autocomplete="off"
          class="el-input__inner"
          rows="2"
          type="text"
          validateevent="true"
        />
        <!---->
        <!---->
        <!---->
      </div>
      <!---->
    </div>
  </div>
   
  <div
    class="el-form-item is-required"
  >
    <label
      class="el-form-item__label"
      for="birthDate"
    >
      Date of birth
    </label>
    <div
      class="el-form-item__content"
    >
      <div
        class="el-date-editor el-input el-input--prefix el-input--suffix el-date-editor--date"
        formatvalue="yyyy-MM-dd"
      >
        <!---->
        <input
          autocomplete="off"
          class="el-input__inner"
          id=""
          name=""
          placeholder="Pick a day"
          prefixicon="el-icon-date"
          rows="2"
          size=""
          type="text"
        />
        <span
          class="el-input__prefix"
        >
          <i
            class="el-input__icon el-icon-date"
          />
        </span>
        <span
          class="el-input__suffix"
        >
          <span
            class="el-input__suffix-inner"
          >
            <i
              class="el-input__icon"
            />
            <!---->
          </span>
          <!---->
        </span>
        <!---->
      </div>
      <!---->
    </div>
  </div>
   
  <div
    class="el-form-item is-required"
  >
    <!---->
    <div
      class="el-form-item__content"
    >
      <label
        class="el-form-item__label no-float"
      >
        E-mails
      </label>
       
      <div
        class="el-form-item is-required"
      >
        <!---->
        <div
          class="el-form-item__content"
        >
          <div
            class="email-input el-input"
          >
            <!---->
            <input
              autocomplete="off"
              class="el-input__inner"
              rows="2"
              type="text"
              validateevent="true"
            />
            <!---->
            <!---->
            <!---->
          </div>
           
          <span
            class="delete-item"
          >
            <svg
              class="fa-icon"
              color="red"
              height="32"
              role="presentation"
              style="font-size: 2em;"
              version="1.1"
              viewBox="0 0 1536 1792"
              width="27.428571428571427"
            >
              <path
                d="M1216 960v-128q0-26-19-45t-45-19h-768q-26 0-45 19t-19 45v128q0 26 19 45t45 19h768q26 0 45-19t19-45zM1536 896q0 209-103 385.5t-279.5 279.5-385.5 103-385.5-103-279.5-279.5-103-385.5 103-385.5 279.5-279.5 385.5-103 385.5 103 279.5 279.5 103 385.5z"
              />
               
    
              <!---->
            </svg>
          </span>
          <!---->
        </div>
      </div>
      <div
        class="el-form-item is-required"
      >
        <!---->
        <div
          class="el-form-item__content"
        >
          <div
            class="email-input el-input"
          >
            <!---->
            <input
              autocomplete="off"
              class="el-input__inner"
              rows="2"
              type="text"
              validateevent="true"
            />
            <!---->
            <!---->
            <!---->
          </div>
           
          <span
            class="delete-item"
          >
            <svg
              class="fa-icon"
              color="red"
              height="32"
              role="presentation"
              style="font-size: 2em;"
              version="1.1"
              viewBox="0 0 1536 1792"
              width="27.428571428571427"
            >
              <path
                d="M1216 960v-128q0-26-19-45t-45-19h-768q-26 0-45 19t-19 45v128q0 26 19 45t45 19h768q26 0 45-19t19-45zM1536 896q0 209-103 385.5t-279.5 279.5-385.5 103-385.5-103-279.5-279.5-103-385.5 103-385.5 279.5-279.5 385.5-103 385.5 103 279.5 279.5 103 385.5z"
              />
               
    
              <!---->
            </svg>
          </span>
          <!---->
        </div>
      </div>
       
      <div
        class="add-item"
      >
        <svg
          class="fa-icon"
          color="green"
          height="32"
          role="presentation"
          style="font-size: 2em;"
          version="1.1"
          viewBox="0 0 1536 1792"
          width="27.428571428571427"
        >
          <path
            d="M1216 960v-128q0-26-19-45t-45-19h-256v-256q0-26-19-45t-45-19h-128q-26 0-45 19t-19 45v256h-256q-26 0-45 19t-19 45v128q0 26 19 45t45 19h256v256q0 26 19 45t45 19h128q26 0 45-19t19-45v-256h256q26 0 45-19t19-45zM1536 896q0 209-103 385.5t-279.5 279.5-385.5 103-385.5-103-279.5-279.5-103-385.5 103-385.5 279.5-279.5 385.5-103 385.5 103 279.5 279.5 103 385.5z"
          />
           
    
          <!---->
        </svg>
      </div>
      <!---->
    </div>
  </div>
   
  <fieldset>
    <legend>
      Addresses
    </legend>
     
    <div>
      <span
        class="delete-address"
      >
        <svg
          class="fa-icon"
          color="red"
          height="32"
          role="presentation"
          style="font-size: 2em;"
          version="1.1"
          viewBox="0 0 1536 1792"
          width="27.428571428571427"
        >
          <path
            d="M1216 960v-128q0-26-19-45t-45-19h-768q-26 0-45 19t-19 45v128q0 26 19 45t45 19h768q26 0 45-19t19-45zM1536 896q0 209-103 385.5t-279.5 279.5-385.5 103-385.5-103-279.5-279.5-103-385.5 103-385.5 279.5-279.5 385.5-103 385.5 103 279.5 279.5 103 385.5z"
          />
           
    
          <!---->
        </svg>
      </span>
       
      <fieldset>
        <div
          class="el-form-item is-required"
        >
          <label
            class="el-form-item__label"
            for="addresses.0.street"
          >
            Street
          </label>
          <div
            class="el-form-item__content"
          >
            <div
              class="el-input"
            >
              <!---->
              <input
                autocomplete="off"
                class="el-input__inner"
                rows="2"
                type="text"
                validateevent="true"
              />
              <!---->
              <!---->
              <!---->
            </div>
            <!---->
          </div>
        </div>
         
        <div
          class="inline"
        >
          <div
            class="el-form-item is-required"
          >
            <label
              class="el-form-item__label"
              for="addresses.0.city"
            >
              City
            </label>
            <div
              class="el-form-item__content"
            >
              <div
                class="el-input"
              >
                <!---->
                <input
                  autocomplete="off"
                  class="el-input__inner"
                  rows="2"
                  type="text"
                  validateevent="true"
                />
                <!---->
                <!---->
                <!---->
              </div>
              <!---->
            </div>
          </div>
           
          <div
            class="el-form-item"
          >
            <label
              class="el-form-item__label"
            >
              State
            </label>
            <div
              class="el-form-item__content"
            >
              <div
                class="el-input"
              >
                <!---->
                <input
                  autocomplete="off"
                  class="el-input__inner"
                  rows="2"
                  type="text"
                  validateevent="true"
                />
                <!---->
                <!---->
                <!---->
              </div>
              <!---->
            </div>
          </div>
           
          <div
            class="el-form-item"
          >
            <label
              class="el-form-item__label"
            >
              Zipcode
            </label>
            <div
              class="el-form-item__content"
            >
              <div
                class="el-input"
              >
                <!---->
                <input
                  autocomplete="off"
                  class="el-input__inner"
                  rows="2"
                  type="text"
                  validateevent="true"
                />
                <!---->
                <!---->
                <!---->
              </div>
              <!---->
            </div>
          </div>
        </div>
      </fieldset>
    </div>
     
    <div
      class="add-item"
    >
      <svg
        class="fa-icon"
        color="green"
        height="32"
        role="presentation"
        style="font-size: 2em;"
        version="1.1"
        viewBox="0 0 1536 1792"
        width="27.428571428571427"
      >
        <path
          d="M1216 960v-128q0-26-19-45t-45-19h-256v-256q0-26-19-45t-45-19h-128q-26 0-45 19t-19 45v256h-256q-26 0-45 19t-19 45v128q0 26 19 45t45 19h256v256q0 26 19 45t45 19h128q26 0 45-19t19-45v-256h256q26 0 45-19t19-45zM1536 896q0 209-103 385.5t-279.5 279.5-385.5 103-385.5-103-279.5-279.5-103-385.5 103-385.5 279.5-279.5 385.5-103 385.5 103 279.5 279.5 103 385.5z"
        />
         
    
        <!---->
      </svg>
    </div>
  </fieldset>
   
  <div
    class="el-form-item is-required"
  >
    <!---->
    <div
      class="el-form-item__content"
    >
      <label
        class="el-form-item__label no-float"
      >
        Phone numbers
      </label>
       
      <div
        class="el-form-item is-required"
      >
        <!---->
        <div
          class="el-form-item__content"
        >
          <div
            class="email-input el-input"
          >
            <!---->
            <input
              autocomplete="off"
              class="el-input__inner"
              rows="2"
              type="text"
              validateevent="true"
            />
            <!---->
            <!---->
            <!---->
          </div>
           
          <span
            class="delete-item"
          >
            <svg
              class="fa-icon"
              color="red"
              height="32"
              role="presentation"
              style="font-size: 2em;"
              version="1.1"
              viewBox="0 0 1536 1792"
              width="27.428571428571427"
            >
              <path
                d="M1216 960v-128q0-26-19-45t-45-19h-768q-26 0-45 19t-19 45v128q0 26 19 45t45 19h768q26 0 45-19t19-45zM1536 896q0 209-103 385.5t-279.5 279.5-385.5 103-385.5-103-279.5-279.5-103-385.5 103-385.5 279.5-279.5 385.5-103 385.5 103 279.5 279.5 103 385.5z"
              />
               
    
              <!---->
            </svg>
          </span>
          <!---->
        </div>
      </div>
      <div
        class="el-form-item is-required"
      >
        <!---->
        <div
          class="el-form-item__content"
        >
          <div
            class="email-input el-input"
          >
            <!---->
            <input
              autocomplete="off"
              class="el-input__inner"
              rows="2"
              type="text"
              validateevent="true"
            />
            <!---->
            <!---->
            <!---->
          </div>
           
          <span
            class="delete-item"
          >
            <svg
              class="fa-icon"
              color="red"
              height="32"
              role="presentation"
              style="font-size: 2em;"
              version="1.1"
              viewBox="0 0 1536 1792"
              width="27.428571428571427"
            >
              <path
                d="M1216 960v-128q0-26-19-45t-45-19h-768q-26 0-45 19t-19 45v128q0 26 19 45t45 19h768q26 0 45-19t19-45zM1536 896q0 209-103 385.5t-279.5 279.5-385.5 103-385.5-103-279.5-279.5-103-385.5 103-385.5 279.5-279.5 385.5-103 385.5 103 279.5 279.5 103 385.5z"
              />
               
    
              <!---->
            </svg>
          </span>
          <!---->
        </div>
      </div>
       
      <div
        class="add-item"
      >
        <svg
          class="fa-icon"
          color="green"
          height="32"
          role="presentation"
          style="font-size: 2em;"
          version="1.1"
          viewBox="0 0 1536 1792"
          width="27.428571428571427"
        >
          <path
            d="M1216 960v-128q0-26-19-45t-45-19h-256v-256q0-26-19-45t-45-19h-128q-26 0-45 19t-19 45v256h-256q-26 0-45 19t-19 45v128q0 26 19 45t45 19h256v256q0 26 19 45t45 19h128q26 0 45-19t19-45v-256h256q26 0 45-19t19-45zM1536 896q0 209-103 385.5t-279.5 279.5-385.5 103-385.5-103-279.5-279.5-103-385.5 103-385.5 279.5-279.5 385.5-103 385.5 103 279.5 279.5 103 385.5z"
          />
           
    
          <!---->
        </svg>
      </div>
      <!---->
    </div>
  </div>
</form>
`;

exports[`ContactInputFields.vue should render the content correctly when no contact is given 1`] = `
<form
  class="el-form"
>
  <div
    class="el-form-item is-required"
  >
    <label
      class="el-form-item__label"
      for="firstname"
    >
      First name
    </label>
    <div
      class="el-form-item__content"
    >
      <div
        class="el-input"
      >
        <!---->
        <input
          autocomplete="off"
          class="el-input__inner"
          rows="2"
          type="text"
          validateevent="true"
        />
        <!---->
        <!---->
        <!---->
      </div>
      <!---->
    </div>
  </div>
   
  <div
    class="el-form-item is-required"
  >
    <label
      class="el-form-item__label"
      for="lastname"
    >
      Last name
    </label>
    <div
      class="el-form-item__content"
    >
      <div
        class="el-input"
      >
        <!---->
        <input
          autocomplete="off"
          class="el-input__inner"
          rows="2"
          type="text"
          validateevent="true"
        />
        <!---->
        <!---->
        <!---->
      </div>
      <!---->
    </div>
  </div>
   
  <div
    class="el-form-item is-required"
  >
    <label
      class="el-form-item__label"
      for="birthDate"
    >
      Date of birth
    </label>
    <div
      class="el-form-item__content"
    >
      <div
        class="el-date-editor el-input el-input--prefix el-input--suffix el-date-editor--date"
        formatvalue="yyyy-MM-dd"
      >
        <!---->
        <input
          autocomplete="off"
          class="el-input__inner"
          id=""
          name=""
          placeholder="Pick a day"
          prefixicon="el-icon-date"
          rows="2"
          size=""
          type="text"
        />
        <span
          class="el-input__prefix"
        >
          <i
            class="el-input__icon el-icon-date"
          />
        </span>
        <span
          class="el-input__suffix"
        >
          <span
            class="el-input__suffix-inner"
          >
            <i
              class="el-input__icon"
            />
            <!---->
          </span>
          <!---->
        </span>
        <!---->
      </div>
      <!---->
    </div>
  </div>
   
  <div
    class="el-form-item is-required"
  >
    <!---->
    <div
      class="el-form-item__content"
    >
      <label
        class="el-form-item__label no-float"
      >
        E-mails
      </label>
        
      <div
        class="add-item"
      >
        <svg
          class="fa-icon"
          color="green"
          height="32"
          role="presentation"
          style="font-size: 2em;"
          version="1.1"
          viewBox="0 0 1536 1792"
          width="27.428571428571427"
        >
          <path
            d="M1216 960v-128q0-26-19-45t-45-19h-256v-256q0-26-19-45t-45-19h-128q-26 0-45 19t-19 45v256h-256q-26 0-45 19t-19 45v128q0 26 19 45t45 19h256v256q0 26 19 45t45 19h128q26 0 45-19t19-45v-256h256q26 0 45-19t19-45zM1536 896q0 209-103 385.5t-279.5 279.5-385.5 103-385.5-103-279.5-279.5-103-385.5 103-385.5 279.5-279.5 385.5-103 385.5 103 279.5 279.5 103 385.5z"
          />
           
    
          <!---->
        </svg>
      </div>
      <!---->
    </div>
  </div>
   
  <fieldset>
    <legend>
      Addresses
    </legend>
      
    <div
      class="add-item"
    >
      <svg
        class="fa-icon"
        color="green"
        height="32"
        role="presentation"
        style="font-size: 2em;"
        version="1.1"
        viewBox="0 0 1536 1792"
        width="27.428571428571427"
      >
        <path
          d="M1216 960v-128q0-26-19-45t-45-19h-256v-256q0-26-19-45t-45-19h-128q-26 0-45 19t-19 45v256h-256q-26 0-45 19t-19 45v128q0 26 19 45t45 19h256v256q0 26 19 45t45 19h128q26 0 45-19t19-45v-256h256q26 0 45-19t19-45zM1536 896q0 209-103 385.5t-279.5 279.5-385.5 103-385.5-103-279.5-279.5-103-385.5 103-385.5 279.5-279.5 385.5-103 385.5 103 279.5 279.5 103 385.5z"
        />
         
    
        <!---->
      </svg>
    </div>
  </fieldset>
   
  <div
    class="el-form-item is-required"
  >
    <!---->
    <div
      class="el-form-item__content"
    >
      <label
        class="el-form-item__label no-float"
      >
        Phone numbers
      </label>
        
      <div
        class="add-item"
      >
        <svg
          class="fa-icon"
          color="green"
          height="32"
          role="presentation"
          style="font-size: 2em;"
          version="1.1"
          viewBox="0 0 1536 1792"
          width="27.428571428571427"
        >
          <path
            d="M1216 960v-128q0-26-19-45t-45-19h-256v-256q0-26-19-45t-45-19h-128q-26 0-45 19t-19 45v256h-256q-26 0-45 19t-19 45v128q0 26 19 45t45 19h256v256q0 26 19 45t45 19h128q26 0 45-19t19-45v-256h256q26 0 45-19t19-45zM1536 896q0 209-103 385.5t-279.5 279.5-385.5 103-385.5-103-279.5-279.5-103-385.5 103-385.5 279.5-279.5 385.5-103 385.5 103 279.5 279.5 103 385.5z"
          />
           
    
          <!---->
        </svg>
      </div>
      <!---->
    </div>
  </div>
</form>
`;
